File: problems/daily/Python3/70. Climbing Stairs.py

class Solution:
    def climbStairs(self, n: int) -> int:
        dp = [0 for _ in range(n+1)]
        dp[0] = 1

        for i in range(n):
            dp[i+1] += dp[i]
            if i + 2 < n + 1:
                dp[i+2] += dp[i]
        return dp[n]

File: problems/daily/Ruby/70. Climbing Stairs.rb
# @param {Integer} n
# @return {Integer}
def climb_stairs(n)
    ans = Array.new(n+1,0)
    ans[0] = 1

    (0...n).each do |i|
        ans[i+1] += ans[i]
        if i + 2 < n +1
            ans[i+2] += ans[i]
        end
    end

    return ans[n]
end